### PR TITLE
Activity manager performance improvements

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Some methods are called quite often with the same values and the results can not change, so we can "cache" the result for the page call, e.g. `getSpecialParameterList()` is now only called once instead of 120-times in case the 30 activities of the same type are displayed :-X
- [x] Check whether filters are valid, before querying extensions about them.
- [x] Ordered the methods of the extension to be more logically related.

PS: Look at the individual commits, to easily see that the methods have not changed their signature, but have only been moved.

/cc @LukasReschke @DeepDiver1975

### CODE REVIEW

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

File: lib/private/activitymanager.php [hunk 1]
```diff
@@ -173,38 +192,30 @@ function translate($app, $text, $params, $stripPath, $highlightParams, $language
 	 * @return array|false
 	 */
 	function getSpecialParameterList($app, $text) {
```
Comment (PVince81) [inline lib/private/activitymanager.php]: What's a "special parameter" by the way ? The PHPDoc doesn't help :smile:

File: lib/private/activitymanager.php [hunk 2]
```diff
@@ -249,23 +260,54 @@ function getNavigation() {
...
+		if (!$this->isFilterValid($filter)) {
+			return $types;
+		}
+
+		foreach($this->extensions as $extension) {
+			$c = $extension();
+			if ($c instanceof IExtension) {
+				$result = $c->filterNotificationTypes($types, $filter);
+				if (is_array($result)) {
+					$types = $result;
```
Comment (PVince81) [inline lib/private/activitymanager.php]: Add `break;` ?

File: lib/private/activitymanager.php (hunk 1 above)
Comment (PVince81) [inline lib/private/activitymanager.php]: Never mind, found in the interface def.

File: lib/private/activitymanager.php (hunk 2 above)
Comment (nickvergessen) [inline lib/private/activitymanager.php]: No, multiple apps might want to modify a default filter (`all`, `by`, `self` from the activity app itself)

File: lib/private/activitymanager.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/activitymanager.php]: Documentation will be fixed in another PR

File: lib/private/activitymanager.php (hunk 2 above)
Comment (PVince81) [inline lib/private/activitymanager.php]: So this means that if multiple extensions/apps modify the default filter, it will only take the last one, right ? (by design)

Comment (nickvergessen) [inline lib/private/activitymanager.php]: No, types is an array. The idea is that e.g. files app removes one type from the filter, files_sharing addes one, etc.
In the final result all changes should be there

Comment (PVince81) [inline lib/private/activitymanager.php]: Ok, makes sense. I didn't see that `$types` was re-passed to the next handler.

**Comment by PVince81** [comment]
:+1:

**Comment by LukasReschke** [comment]
:+1:
